timestampvm/block.go

```diff
@@ -5,52 +5,71 @@ package timestampvm
 
 import (
 	"errors"
+	"fmt"
 	"time"
 
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow/choices"
 	"github.com/ava-labs/avalanchego/snow/consensus/snowman"
-	"github.com/ava-labs/avalanchego/vms/components/core"
+	"github.com/ava-labs/avalanchego/utils/hashing"
 )
 
 var (
 	errTimestampTooEarly = errors.New("block's timestamp is earlier than its parent's timestamp")
 	errDatabaseGet       = errors.New("error while retrieving data from database")
-	errDatabaseSave      = errors.New("error while saving block to the database")
 	errTimestampTooLate  = errors.New("block's timestamp is more than 1 hour ahead of local time")
-	errBlockType         = errors.New("unexpected block type")
+	errBlockNil          = errors.New("block is nil")
 
-	_ snowman.Block = &Block{}
+	_ Block = &TimeBlock{}
 )
 
+// Block interface is a wrapper around snowman.Block
+type Block interface {
```

Comment:
Is this interface actually useful? I suppose maybe for mocking with the state tests? As it is right now I don't think we use it for anything.

Reply:
it is not very useful indeed. just followed same logic with other VM blocks. gonna remove this.